### DISCUSSION TITLE
Are there any efforts for testground on TypeScript?

### DISCUSSION THREAD

**Discussion by 0xjjpa** [testground/testground#1086] [General] [answered]
👋 ! First of all, congrats on Testground, it's looking great! 

Quick question here, exploring the codebase and seeing the existing plans ([libp2p](https://github.com/libp2p/test-plans) and [ipfs](https://github.com/ipfs/test-plans)) it seems that Testground is mostly focused in Go and Go-based implementations of `libp2p` right now. However, in the [May blog post from IPFS](https://blog.ipfs.io/2020-05-06-launching-testground/) there's a mention of a community effort to port an SDK in TypeScript for `node` and `browser`. Is that still the case? If so, do you have any active repository or work publicly available going on that direction? Latest relevant ticket seems to be https://github.com/testground/testground/issues/137, but seems a bit old. 

We are using `js-libp2p` in our [privacy oriented p2p protocol](https://github.com/hoprnet/hopr-core), so it would be great to work together on Testground's tooling for JavaScript/TypeScript-oriented projects that use `js-libp2p`, as were are facing similar [problems](https://github.com/hoprnet/hopr-core/issues/104) when testing/simulating our network.

**Comment by raulk** [comment]
Hey! Absolutely. We did have a TypeScript SDK as a wishlist item for Q2 (via @hugomrdias), but we weren’t able to land it due to other priorities. And to my knowledge there has been no substantial progress on the implementation side of the equation. But we did discuss design details. Our intention was to create node and a browser variants, using the webdis library to implement the Redis calls in the sync service. Maybe Hugo can elaborate a little more. And if you’re interested in picking this up, it would be awesome!

**Reply by 0xjjpa** [reply]
Oh nice! Would be more than happy to work on that (also up for collaboration with @hugomrdias, used https://github.com/moxystudio/next-with-moxy in the past, huge fan of Moxy). We can even push it as a submission to [HackFs](https://hackfs.com/). We anyway will use something similar for our own [project](https://github.com/hoprnet/hopr-core), although for now we are focusing in the `node` variant, and might move away from TCP to [QUIC](https://github.com/alanshaw/js-libp2p-quic). Either way, let me know here or via DM (@jjperezaguinaga in Twitter) your thoughts on the best ways to pool efforts here.

**Reply by hugomrdias** [reply]
This would be awesome for HackFS, @raulk what do you think ? I will probably not have time to fully work on this but i can definitively support you @jjperezaguinaga

**Reply by 0xjjpa** [reply]
Sounds good! I'm already in the Slack channel so you can just ping me there under `Jose Aguinaga`

**Reply by hugomrdias** [reply]
Which channel ?

**Reply by 0xjjpa** [reply]
In the [HackFS Filecoin](filecoinproject.slack.com) workspace; I'm unaware whether there's another one specific for `libp2p` or `ipfs`, but you can always just reach out/invite me through `jose.aguinaga@hoprnet.io`, whatever works for you.

**Comment by 0xjjpa** [comment, marked as answer]
@raulk I managed to get in touch with @hugomrdias and it seems that as of right now, there will not be immediate support for Testground on TypeScript. My team will continue the possibility of pushing on that direction, and if we stumble upon any issues, Hugo was kind enough to offer some assistance. We can close this conversation for now, thanks all!

**Comment by hacdias** [comment]
We're starting to work on a version of the [JavaScript SDK](https://github.com/testground/sdk-js)!

**Reply by 0xjjpa** [reply]
@hacdias Yay! I _knew_ it would pay out to check back this discussion. We are back again facing the same issue in our project (https://github.com/hoprnet/hoprnet/issues/111) so naturally was checking this again. Do you have a roadmap or milestone or anything we can subscribe ourselves to other than the repository?
